### PR TITLE
fix: only fetch all assocations for flagged applications

### PR DESCRIPTION
**Pull request by lucasmarshall**

### CODE REVIEW

File: packages/core/remotes/impl/hubspot/index.ts [hunk 1]
```diff
@@ -649,6 +655,17 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
     return { standardObjectTypes, customObjectSchemas };
   }
 
+  async #getAssociatedObjectTypesForObjectTypeFeatureFlagged(fromObjectTypeId: string): Promise<{
+    standardObjectTypes: string[];
+    customObjectSchemas: HubSpotCustomSchema[];
+  }> {
+    if (this.#config.applicationId in FETCH_ASSOCIATIONS_APPLICATION_IDS) {
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:662]: does this do the right thing in TS? i just tried:
```
> 'a' in ['a', 'b']
false
```

File: packages/core/remotes/impl/hubspot/index.ts [hunk 2]
```diff
@@ -99,6 +99,11 @@ import {
 const HUBSPOT_RECORD_LIMIT = 100;
 const HUBSPOT_SEARCH_RESULTS_LIMIT = 10000;
 
+// TODO move this to lekko
+const FETCH_ASSOCIATIONS_APPLICATION_IDS = [
+  '9773053e-a13f-4249-b641-301a51952708', // kloze
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:104]: pls remove comment (customer name)

File: packages/core/remotes/impl/hubspot/index.ts (hunk 1 above)
Comment (lucasmarshall) [reply packages/core/remotes/impl/hubspot/index.ts:662]: Yeah, you are right. Will fix.

Comment (lucasmarshall) [reply packages/core/remotes/impl/hubspot/index.ts:662]: Got confused with python 😬

**Review by tomkit — APPROVED** [review]
Does this create an empty key in the interface we hand off to customers and did we do that prior to fetching associations?

I'm thinking about whether this might cause confusion to developers consuming the response. @asdfryan might know more

**Comment by lucasmarshall** [comment]
> Does this create an empty key in the interface we hand off to customers and did we do that prior to fetching associations?
> 
> I'm thinking about whether this might cause confusion to developers consuming the response. @asdfryan might know more

https://github.com/supaglue-labs/supaglue/blob/f71c08896696226a7c746f5f8689d92abcf6f20a/packages/core/remotes/impl/hubspot/index.ts#L826-L827

File: packages/core/remotes/impl/hubspot/index.ts [hunk 3]
```diff
@@ -1207,7 +1222,7 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
   public async getAccount(id: string, fieldMappingConfig: FieldMappingConfig): Promise<Account> {
     const properties = await this.getCommonObjectPropertyIdsToFetch('company');
     const { standardObjectTypes: associatedStandardObjectTypes, customObjectSchemas: associatedCustomObjectSchemas } =
-      await this.#getAssociatedObjectTypesForObjectType('deal');
+      await this.#getAssociatedObjectTypesForObjectTypeFeatureFlagged('deal');
```
Comment (asdfryan) [inline packages/core/remotes/impl/hubspot/index.ts:1225]: Hmm this doesn't make sense to me. Shouldn't this be `company`?

File: packages/core/remotes/impl/hubspot/index.ts [hunk 4]
```diff
@@ -1577,7 +1592,7 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
   public async getContact(id: string, fieldMappingConfig: FieldMappingConfig): Promise<Contact> {
     const properties = await this.getCommonObjectPropertyIdsToFetch('contact');
     const { standardObjectTypes: associatedStandardObjectTypes, customObjectSchemas: associatedCustomObjectSchemas } =
-      await this.#getAssociatedObjectTypesForObjectType('deal');
+      await this.#getAssociatedObjectTypesForObjectTypeFeatureFlagged('deal');
```
Comment (asdfryan) [inline packages/core/remotes/impl/hubspot/index.ts:1595]: This should be `contact`?

Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:1595]: contact?

**Review by asdfryan — APPROVED** [review]

**Pull request merged**